src/logic/equiv/fin.lean

```diff
@@ -424,6 +425,40 @@ def fin_prod_fin_equiv : fin m × fin n ≃ fin (m * n) :=
         ... = y.1 : nat.mod_eq_of_lt y.2),
   right_inv := λ x, fin.eq_of_veq $ nat.mod_add_div _ _ }
 
+/-- The equivalence between `a` and `(a / n, a % n)` for nonzero `n`.
+
+This is like `fin_prod_fin_equiv.symm` but with `m` infinite.
+See `nat.div_mod_unique` for a similar propositional statement. -/
+@[simps]
+def nat.div_mod_equiv (n : ℕ) [ne_zero n] : ℕ ≃ ℕ × fin n :=
+{ to_fun := λ a, (a / n, fin.of_nat' a),
```

Comment:
You can use the simp normal form (coercion) here, and keep the same proof:
```suggestion
{ to_fun := λ a, (a / n, a),
```

Reply:
I opted for writing an explicit `↑` to make it clear that there's something invisible happening here (the mod operation)